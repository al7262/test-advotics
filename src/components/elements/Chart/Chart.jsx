import React from 'react';
// import PropTypes from 'prop-types';
import {
  ComposedChart,
  Line,
  Bar,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  Legend,
  ResponsiveContainer,
} from 'recharts';
import styles from './index.css';

const data = [
  {
    name: 'Page A',
    uv: 590,
    pv: 800,
    amt: 1400,
    cnt: 490,
  },
  {
    name: 'Page B',
    uv: 868,
    pv: 967,
    amt: 1506,
    cnt: 590,
  },
  {
    name: 'Page C',
    uv: 1397,
    pv: 1098,
    amt: 989,
    cnt: 350,
  },
  {
    name: 'Page D',
    uv: 1480,
    pv: 1200,
    amt: 1228,
    cnt: 480,
  },
  {
    name: 'Page E',
    uv: 1520,
    pv: 1108,
    amt: 1100,
    cnt: 460,
  },
  {
    name: 'Page F',
    uv: 1400,
    pv: 680,
    amt: 1700,
    cnt: 380,
  },
];

export default function Chart() {
  return (
    <div className={styles.root}>
      <ResponsiveContainer width="100%" height="100%">
        <ComposedChart
          width={600}
          height={400}
          data={data}
          margin={{
            top: 16,
            right: 0,
            bottom: 16,
            left: -16,
          }}
        >
          <CartesianGrid stroke="#f5f5f5" />
          <XAxis dataKey="name" scale="band" />
          <YAxis />
          <Tooltip />
          <Legend />
          <Bar dataKey="pv" barSize={20} fill="var(--primary-color)" />
          <Bar dataKey="amt" barSize={20} fill="var(--secondary-color-light)" />
          <Bar dataKey="cnt" barSize={20} fill="var(--red-color)" />
          <Line type="monotone" dataKey="uv" stroke="var(--secondary-color)" strokeWidth={2} />
        </ComposedChart>
      </ResponsiveContainer>
    </div>
  );
}

Chart.defaultProps = {
};

Chart.propTypes = {
};
