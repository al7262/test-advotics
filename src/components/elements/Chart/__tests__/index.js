/* eslint-disable react/jsx-filename-extension */
/* eslint-disable no-undef */
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Chart from '../Chart';

describe('src/components/elements/Chart', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Chart />);
    expect(tree).toMatchSnapshot();
  });
});
