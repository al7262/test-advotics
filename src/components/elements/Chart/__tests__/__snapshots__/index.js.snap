// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Chart render 1`] = `
<div>
  <ResponsiveContainer
    debounce={0}
    height="100%"
    width="100%"
  >
    <ComposedChart
      barCategoryGap="10%"
      barGap={4}
      data={
        Array [
          Object {
            "amt": 1400,
            "cnt": 490,
            "name": "Page A",
            "pv": 800,
            "uv": 590,
          },
          Object {
            "amt": 1506,
            "cnt": 590,
            "name": "Page B",
            "pv": 967,
            "uv": 868,
          },
          Object {
            "amt": 989,
            "cnt": 350,
            "name": "Page C",
            "pv": 1098,
            "uv": 1397,
          },
          Object {
            "amt": 1228,
            "cnt": 480,
            "name": "Page D",
            "pv": 1200,
            "uv": 1480,
          },
          Object {
            "amt": 1100,
            "cnt": 460,
            "name": "Page E",
            "pv": 1108,
            "uv": 1520,
          },
          Object {
            "amt": 1700,
            "cnt": 380,
            "name": "Page F",
            "pv": 680,
            "uv": 1400,
          },
        ]
      }
      height={400}
      layout="horizontal"
      margin={
        Object {
          "bottom": 16,
          "left": -16,
          "right": 0,
          "top": 16,
        }
      }
      reverseStackOrder={false}
      stackOffset="none"
      syncMethod="index"
      width={600}
    >
      <CartesianGrid
        fill="none"
        horizontal={true}
        horizontalFill={Array []}
        horizontalPoints={Array []}
        stroke="#f5f5f5"
        vertical={true}
        verticalFill={Array []}
        verticalPoints={Array []}
      />
      <XAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        dataKey="name"
        domain={
          Array [
            0,
            "auto",
          ]
        }
        height={30}
        hide={false}
        mirror={false}
        orientation="bottom"
        padding={
          Object {
            "left": 0,
            "right": 0,
          }
        }
        reversed={false}
        scale="band"
        tickCount={5}
        type="category"
        width={0}
        xAxisId={0}
      />
      <YAxis
        allowDataOverflow={false}
        allowDecimals={true}
        allowDuplicatedCategory={true}
        domain={
          Array [
            0,
            "auto",
          ]
        }
        height={0}
        hide={false}
        mirror={false}
        orientation="left"
        padding={
          Object {
            "bottom": 0,
            "top": 0,
          }
        }
        reversed={false}
        scale="auto"
        tickCount={5}
        type="number"
        width={60}
        yAxisId={0}
      />
      <Tooltip
        active={false}
        allowEscapeViewBox={
          Object {
            "x": false,
            "y": false,
          }
        }
        animationDuration={400}
        animationEasing="ease"
        contentStyle={Object {}}
        coordinate={
          Object {
            "x": 0,
            "y": 0,
          }
        }
        cursor={true}
        cursorStyle={Object {}}
        filterNull={true}
        isAnimationActive={false}
        itemStyle={Object {}}
        labelStyle={Object {}}
        offset={10}
        separator=" : "
        trigger="hover"
        useTranslate3d={false}
        viewBox={
          Object {
            "x1": 0,
            "x2": 0,
            "y1": 0,
            "y2": 0,
          }
        }
        wrapperStyle={Object {}}
      />
      <Legend
        align="center"
        iconSize={14}
        layout="horizontal"
        verticalAlign="bottom"
      />
      <Bar
        animationBegin={0}
        animationDuration={400}
        animationEasing="ease"
        barSize={20}
        data={Array []}
        dataKey="pv"
        fill="var(--primary-color)"
        hide={false}
        isAnimationActive={false}
        layout="vertical"
        legendType="rect"
        minPointSize={0}
        xAxisId={0}
        yAxisId={0}
      />
      <Bar
        animationBegin={0}
        animationDuration={400}
        animationEasing="ease"
        barSize={20}
        data={Array []}
        dataKey="amt"
        fill="var(--secondary-color-light)"
        hide={false}
        isAnimationActive={false}
        layout="vertical"
        legendType="rect"
        minPointSize={0}
        xAxisId={0}
        yAxisId={0}
      />
      <Bar
        animationBegin={0}
        animationDuration={400}
        animationEasing="ease"
        barSize={20}
        data={Array []}
        dataKey="cnt"
        fill="var(--red-color)"
        hide={false}
        isAnimationActive={false}
        layout="vertical"
        legendType="rect"
        minPointSize={0}
        xAxisId={0}
        yAxisId={0}
      />
      <Line
        activeDot={true}
        animateNewValues={true}
        animationBegin={0}
        animationDuration={1500}
        animationEasing="ease"
        connectNulls={false}
        dataKey="uv"
        dot={true}
        fill="#fff"
        hide={false}
        isAnimationActive={false}
        legendType="line"
        points={Array []}
        stroke="var(--secondary-color)"
        strokeWidth={2}
        type="monotone"
        xAxisId={0}
        yAxisId={0}
      />
    </ComposedChart>
  </ResponsiveContainer>
</div>
`;
